### PR TITLE
Sarkars/test framework

### PR DESCRIPTION
**Pull request by sayantan-nervana**

### CODE REVIEW

File: test/python/test_modeltester.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+class TestModelTester(NgraphTest):
+
+    def test_MLP(self):
+        cwd = os.getcwd()
+        os.chdir('../model_level_tests/')
+        try:
+            command_executor(
+                "python test_main.py --run_logparse_tests --models MLP")
+        finally:
+            os.chdir(cwd)
```
Comment (sayantan-nervana) [inline test/python/test_modeltester.py]: TODO: does this need expansion to test the framework more?

File: test/model_level_tests/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+2. **Disabling test suites or sub-tests**: Using the `--disable` flag you can disable whole test suites or certain subtests of certain test suites. For example: `--disable "MLP.test1,MLP_fail_2"` would disable test suite `MLP_fail_2` and sub-test `MLP.test1` (but it would run other tests in `MLP` test suite). Sub tests can also be disabled if their directory name has `disabled` in it, for example `MLP/test3_disabled`
+3. **Listing tests**: Adding `--list` will produce a list if available tests. It can be coupled with the `--models` and `--disable` flag to get a list of tests that will be run if `--list` is not present. For example the command `python test_main.py --run_logparse_tests --models MLP_custom_logparser,MLP,MLP_fail_1,MLP_fail_2 --list --disable "MLP.test1,MLP_fail_2"` prints the following. <img src="./resources/list.png" height="415" width="350">
```
Comment (kanvi-nervana) [inline test/model_level_tests/README.md]: a list if available  ---->  a list of available

Comment (kanvi-nervana) [reply test/model_level_tests/README.md]: May be add the images on the next line.

File: test/model_level_tests/README.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        1. `README.md` (*Optional*): Information about this `test suite`. Expected to be short as this text is used by `test_main.py` to print a short help on available tests.
+        2. `repo.txt`: Clonable git URL in the first line, Optionally branch/SHA/tag in the second line (by default master will be used)
+        3. `getting_repo_ready.sh` (*Optional*): An executable shell script to run before the tests start running for this repo. Can be used to install prerequisites.
+        4. `enable_ngraph.patch` (*Optional*): A patch to be applied to the downloaded repo (**2.1.2**). If this file exists it is used by all `sub-test` directories (**2.1.5**), unless overridden by their own patch file (**2.1.5.1**). Either a global or a local patch file must be present.
+        5. `Sub-test` (*Repeated*): Should start with the string `test`. Can be disabled by adding the string `disabled` to its name.
+            1. `enable_ngraph.patch` (*Optional*): If a `sub-test` contains its own patch, that takes precedence over the global patch (**2.1.4**)
+            2. `core_rewrite_test.sh`: The main run script for this `sub-test`.
```
Comment (kanvi-nervana) [inline test/model_level_tests/README.md]: core_run_test.sh

File: test/model_level_tests/README.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+### Directory Structure
+This section describes the directory structure. Indentation means sub-directories. Top to bottom, roughly, they are in the order in which they are executed
+1. `test_main.py`: The CLI to trigger different tests.
+2. `models`: The directory containing all test suites.
+    1. `Test suite` or `Model test directory` (*Repeated*): A directory in `models`. Each `test suite` represents an external repo that must be downloaded. Usually that corresponds to a particular network topology (but certain repos might run multiple topologies).
+        1. `README.md` (*Optional*): Information about this `test suite`. Expected to be short as this text is used by `test_main.py` to print a short help on available tests.
+        2. `repo.txt`: Clonable git URL in the first line, Optionally branch/SHA/tag in the second line (by default master will be used)
```
Comment (sayantan-nervana) [inline test/model_level_tests/README.md]:
```suggestion
        2. `repo.txt`: Clonable git URL in the first line, optional branch/SHA/tag in the second line (by default master will be used)
```

File: test/model_level_tests/README.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        3. `getting_repo_ready.sh` (*Optional*): An executable shell script to run before the tests start running for this repo. Can be used to install prerequisites.
+        4. `enable_ngraph.patch` (*Optional*): A patch to be applied to the downloaded repo (**2.1.2**). If this file exists it is used by all `sub-test` directories (**2.1.5**), unless overridden by their own patch file (**2.1.5.1**). Either a global or a local patch file must be present.
+        5. `Sub-test` (*Repeated*): Should start with the string `test`. Can be disabled by adding the string `disabled` to its name.
```
[the remark this answers is not in the document]
Comment (sayantan-nervana) [reply test/model_level_tests/README.md]: yes

**Pull request merged**